instrumentation/otel-webserver-module/src/nginx/ngx_http_opentelemetry_module.c

```diff
@@ -894,6 +1022,113 @@ static OTEL_SDK_STATUS_CODE otel_startInteraction(ngx_http_request_t* r, const c
     }
     return res;
 }
+static void otel_variables_decorator(ngx_http_request_t* r){
```

Comment:
Can we add some comments on what we are trying to achieve here? Because the code seems complex and not very intuitive to understand?